deepfence_agent/tools/apache/scope/probe/kubernetes/reporter.go

```diff
@@ -346,8 +358,41 @@ func (r *Tagger) Tag(rpt report.Report) (report.Report, error) {
 	return rpt, nil
 }
 
-// Report generates a Report containing Container and ContainerImage topologies
+func (r *Reporter) updateReportCache() {
+	k8sReport, err := r.generateReport()
+	if err != nil {
+		logrus.Error(err.Error())
+	} else {
+		r.kubernetesReportCache.Lock()
+		r.kubernetesReportCache.k8sReport = k8sReport
+		r.kubernetesReportCache.Unlock()
+	}
+	ticker := time.NewTicker(5 * time.Minute)
+	for {
+		select {
+		case <-ticker.C:
+			k8sReport, err = r.generateReport()
```

Comment:
Will this function complete before 5* time.Minute all the time ? Are there cases where it will be slow ? Also, what happens if this function always returns error ? Is there a way to notify, bail out and retry after sometime ?

Reply:
If the report generation is slower than 5 minutes, it won't be an error. We would have just kicked off next report generation. We are safe in terms of updating the cache because of lock, in case those two simultaneous reports are finished at the same time.